### PR TITLE
chore: change default lint level to warning and deny warnings in CI

### PR DESCRIPTION
**Pull request by taiki-e**
## Motivation

https://github.com/tokio-rs/tokio/issues/1083#issuecomment-491411976 and https://github.com/rust-lang-nursery/futures-rs/pull/1448#issuecomment-460572701.

### CODE REVIEW

**Comment by taiki-e** [comment]
This PR seems to have caught some warnings that we missed: https://dev.azure.com/tokio-rs/Tokio/_build/results?buildId=1874&view=logs

File: tokio-sync/src/watch.rs [hunk 1]
```diff
@@ -60,14 +60,16 @@ use crate::task::AtomicWaker;
...
-use futures_core::ready;
 use futures_util::future::poll_fn;
-use futures_util::pin_mut;
 use std::ops;
 use std::sync::atomic::AtomicUsize;
 use std::sync::atomic::Ordering::SeqCst;
 use std::sync::{Arc, Mutex, RwLock, RwLockReadGuard, Weak};
 
+#[cfg(feature = "async-traits")]
+use futures_core::ready;
```
Comment (LucioFranco) [inline tokio-sync/src/watch.rs]: why this change?

Comment (taiki-e) [reply tokio-sync/src/watch.rs]: It is a change to fix the warnings mentioned in https://github.com/tokio-rs/tokio/pull/1416#issuecomment-519820575.
https://dev.azure.com/tokio-rs/Tokio/_build/results?buildId=1874
```text
error: unused import: `futures_core::ready`
  --> tokio-sync/src/watch.rs:63:5
   |
63 | use futures_core::ready;
   |     ^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D unused-imports` implied by `-D warnings`

error: unused import: `futures_util::pin_mut`
  --> tokio-sync/src/watch.rs:65:5
   |
65 | use futures_util::pin_mut;
   |     ^^^^^^^^^^^^^^^^^^^^^
```

Comment (LucioFranco) [reply tokio-sync/src/watch.rs]: Ahh I see now 👍

File: azure-pipelines.yml [hunk 2]
```diff
@@ -3,6 +3,7 @@ pr: ["master", "std-future"]
 
 variables:
   nightly: nightly-2019-07-17
+  RUSTFLAGS: -Dwarnings
```
Comment (taiki-e) [inline azure-pipelines.yml]: Oh, I forgot to add this to .cirrus.yml.
